src/deluge/util/functions.h

```diff
@@ -468,6 +468,39 @@ inline void getBlurColour(uint8_t rgb[], uint8_t fromRgb[]) {
 	rgb[2] = ((uint32_t)fromRgb[2] * 1 + averageBrightness) >> 5;
 }
 
+// clang-format off
+
+constexpr uint32_t IMat[4][4] = {{ONE_Q15, 0, 0, 0},
+							{0, ONE_Q15, 0, 0},
+							{0, 0, ONE_Q15, 0},
+							{0, 0, 0, ONE_Q15}};
+
+constexpr float c = cos(1.0f);
+constexpr float s = sin(1.0f);
+constexpr uint32_t RMat[4][4] = {{(uint32_t)(c*ONE_Q15), 		 			 0, (uint32_t)(s*ONE_Q15), 		 0},
+							 	 {(uint32_t)(s*ONE_Q15), (uint32_t)(c*ONE_Q15), 					0, 		 0},
+							 	 {		  			  0, (uint32_t)(s*ONE_Q15), (uint32_t)(c*ONE_Q15), 		 0},
+							 	 {		  			  0, 		 			 0, 					0, ONE_Q15}};
+
+// clang-format on
+
+inline void xformrgb(uint8_t rgb[], uint8_t fromRgb[], const uint32_t mat[4][4]) {
```

Comment:
is there a top in this function name? should be "fromrgb" instead of "formrgb"?

Reply:
Ya